spec/lib/msf/core/auxiliary/rocketmq_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'spec_helper'
+
+RSpec.describe Msf::Auxiliary::Rocketmq do
+  subject do
+    mod = Msf::Auxiliary.new
+    mod.extend(Msf::Exploit::Remote::Tcp)
+    mod.extend(Msf::Auxiliary::Rocketmq)
+    mod
+  end
+
+  describe 'get_rocketmq_version' do
+    context 'correctly looks up id 401 as V4.9.4' do
+      it 'returns that version' do
+        expect(subject.get_rocketmq_version(401)).to eql('V4.9.4')
+      end
+    end
+
+    context 'correctly looks up id 99999 as UNKNOWN.VERSION.ID.99999' do
+      it 'returns that version' do
+        expect(subject.get_rocketmq_version(99999)).to eql('UNKNOWN.VERSION.ID.99999')
+      end
+    end
+  end
+
+  let(:expected_name_server_request) do
+    "\x00\x00\x00\xC7\x00\x00\x00\xC3{\"code\":105,\"extFields\":{\"Signature\":\"/u5P/wZUbhjanu4LM/UzEdo2u2I=\",\"topic\":\"TBW102\",\"AccessKey\":\"rocketmq2\"},\"flag\":0,\"language\":\"JAVA\",\"opaque\":1,\"serializeTypeCurrentRPC\":\"JSON\",\"version\":401}".b
+  end
+
+  let(:expected_name_server_response) do
```

Comment:
The prefix `expected` here is a bit of a misnomer since it's mock data; Potentially:

```suggestion
  let(:mock_name_server_response) do
```